core-processor/src/main/java/io/micronaut/inject/processing/DeclaredBeanElementCreator.java

```diff
@@ -457,11 +473,22 @@ protected void applyConfigurationInjectionIfNecessary(BeanDefinitionVisitor visi
     /**
      * Is inject point method?
      *
-     * @param memberElement The method
+     * @param propertyElement
+     * @param memberElement   The method
      * @return true if it is
      */
-    protected boolean isInjectPointMethod(MemberElement memberElement) {
-        return memberElement.hasDeclaredStereotype(AnnotationUtil.INJECT);
+    protected boolean isInjectPointMethod(PropertyElement propertyElement, MemberElement memberElement) {
+        return memberElement.hasDeclaredStereotype(AnnotationUtil.INJECT) || isInjectableProperty(propertyElement);
+    }
+
+    private static boolean isInjectableProperty(@Nullable PropertyElement propertyElement) {
```

Comment:
Why do we need to have custom property checks? Write method should have the same annotations.